src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,36 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
 
 public class AsyncRequestProcessor {
+    private static final long TIMEOUT = 1000;
+    private static final String USER_DETAILS = "Details for %s";
+
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() ->
+           cache.computeIfAbsent(
```

Comment:
Using 'computeIfAbsent' on the cache is a thread-safe way to handle concurrent processing. However, you should be aware that the 'getUserData' method will block if it's called for the same userId by multiple threads, potentially leading to performance issues if the user data retrieval is a slow operation.